lib/flight_scheduler/schedulers/fifo_scheduler.rb

```diff
@@ -59,6 +59,14 @@ def allocate_jobs
           else
             job
           end
+
+          if max = job.partition.max_time_limit
+            if job.time_limit.nil? || job.time_limit > max
+              job.reason_pending = 'PartitionTimeLimit'
+              next
+            end
+            job
+          end
```

Comment:
This needs to be before the handling of array jobs.  Otherwise, array jobs are broken.  Also the fifo specs don't pass.  I'll push a commit fixing both soon.